tests/unit_tests/test_client.py

```diff
@@ -355,6 +356,26 @@ def test_uc_not_initialized_isenabled():
     )
 
 
+@responses.activate
+def test_uc_dependency(unleash_client):
+    responses.add(responses.POST, URL + REGISTER_URL, json={}, status=202)
+    responses.add(
+        responses.GET,
+        URL + FEATURES_URL,
+        json=MOCK_FEATURE_WITH_DEPENDENCIES_RESPONSE,
+        status=200,
+    )
+    responses.add(responses.POST, URL + METRICS_URL, json={}, status=202)
+
+    unleash_client.initialize_client()
+    time.sleep(1)
```

Comment:
If we gotta then we gotta but I'd like to avoid tests that use sleep like this. It's making the whole suite crazy slow.  There's some examples in `tests/specification_tests/test_client_specs.py` where we use bootstrapping over responses to achieve the same thing, which avoids the sleep